pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -1171,9 +1151,56 @@ impl<T: Config> Pallet<T> {
 		)
 	}
 
+	/// Consumes the `CollectedInvestment` amounts and
+	/// `CollectedForeignInvestmentHook` notification such that any
+	/// potential consumer could act upon that, e.g. Liquidity Pools for
+	/// `ExecutedCollectInvest`.
+	///
+	/// NOTE: Converts the collected pool currency payment amount to foreign
+	/// currency via the `CurrencyConverter` trait.
+	pub(crate) fn notify_executed_collect_invest(
+		who: &T::AccountId,
+		investment_id: T::InvestmentId,
+	) -> DispatchResult {
+		let foreign_payout_currency = InvestmentPaymentCurrency::<T>::get(who, investment_id)
+			.ok_or(Error::<T>::InvestmentPaymentCurrencyNotFound)?;
+		let pool_currency = T::PoolInspect::currency_for(investment_id.of_pool())
+			.ok_or(Error::<T>::PoolNotFound)?;
+		let collected = CollectedInvestment::<T>::take(who, investment_id);
```

Comment:
Due to being a `ValueQuery` this will always hit. Are we sure the solidity side wants to be informed if nothing was acutally transferred? 

Due to the need of bridging this seems like a big overhead for no information gain.

Reply:
I am confused, in #1473 [you requested](https://github.com/centrifuge/centrifuge-chain/pull/1473#discussion_r1323224830) to allow collecting zero amounts, so I removed the barrier. IIUC, now you contradict your previous request?

Reply:
No, I am not. I am just saying that if we trigger a return message, we should be sure Solidity actually needs it. If not we should collect and not trigger a return message. But lets just keep it as is. Its just optimisation.

Reply:
I have been favouring blocking a return message if the amount is zero. I will implement this.

Reply:
I think there is a possible scenario in which someone submitted a redemption order, and then the price ended up going to 0, and the redemption is fully executed but with 0 currency payout. In that case, the value of sending `ExecutedCollectRedeem` to the other domains is that they will update the local `remaining_redeem_order` state to 0.

Reply:
> I think there is a possible scenario in which someone submitted a redemption order, and then the price ended up going to 0, and the redemption is fully executed but with 0 currency payout. In that case, the value of sending `ExecutedCollectRedeem` to the other domains is that they will update the local `remaining_redeem_order` state to 0.

IIUC, the payment amount should then not be zero. I added a short circuit if the both the accumulated payment amount as well as collected amount are zero: [0b6924d](https://github.com/centrifuge/centrifuge-chain/pull/1556/commits/0b6924df3d14d6c18a15f760746259baca6b47c2)